### PR TITLE
Added google authentication prompt at log in

### PR DESCRIPTION
**Pull request by Olivernyu**

### CODE REVIEW

File: server/app.py [hunk 1]
```diff
@@ -3,17 +3,46 @@
...
+def login():
+    authorizationUrl, state = flow.authorization_url()
+    return redirect(authorizationUrl)
+
+@app.route("/loggedin")
+def loggedin():
+    # flow.fetch_token(authorization_response=request.url)
+    return "Log in successful!"
+
+
```
Comment (leo4life2) [inline server/app.py]: Let's not put endpoints in the app.py file. This file should just contain the Index class that returns the home page. Let's put all the endpoints in the endpoint folder, create classes for them, and import them here. See the `Endpoints` and `HelloWorld` endpoints below as an example. You can just do another `add_resource` when you're done with that.

File: server/app.py [hunk 2]
```diff
@@ -3,17 +3,46 @@
...
 
 app = Flask(__name__, static_url_path='', static_folder='frontend/build')
 api = Api(app)
 
+
+GOOGLE_CLIENT_ID_TEST = "497541279341-qtudp4uvo0g39s0o4ops0mr2dsvemnp5.apps.googleusercontent.com"
+CLIENT_SECRET_FILE = os.path.join(pathlib.Path(__file__).parent, "client_secret_test.json")
+flow = Flow.from_client_secrets_file(
+    client_secrets_file=CLIENT_SECRET_FILE,
+    scopes=["https://www.googleapis.com/auth/userinfo.profile", "https://www.googleapis.com/auth/userinfo.email", "openid"],
```
Comment (leo4life2) [inline server/app.py]: Same with this. We can have these constants and variables in the endpoint's file.

File: server/app.py [hunk 3]
```diff
@@ -3,17 +3,46 @@
 The endpoint called `endpoints` will return all available endpoints.
 """
 
-from flask import Flask, send_from_directory
+import os
+import pathlib
+from flask import Flask, send_from_directory, redirect, request
 from flask_restful import Api, Resource
 from .src.endpoints.endpoints import Endpoints, HelloWorld
+from google_auth_oauthlib.flow import Flow
```
Comment (leo4life2) [inline server/app.py]: Same with the imports. We can do the imports in the endpoint's file.

**Review by leo4life2 — CHANGES_REQUESTED** [review]

File: server/src/endpoints/endpoints.py [hunk 4]
```diff
@@ -1,4 +1,8 @@
+import os
+import pathlib
+from flask import Flask, redirect
 from flask_restx import Resource
+from google_auth_oauthlib.flow import Flow
```
Comment (leo4life2) [inline server/src/endpoints/endpoints.py]: this prob isn't needed here?

Comment (leo4life2) [reply server/src/endpoints/endpoints.py]: and can you get rid of the hello word here as well haha tyty

**Review by leo4life2 — CHANGES_REQUESTED** [review]

**Review by TomY-Zhang — APPROVED** [review]

File: server/src/endpoints/endpoints.py (hunk 4 above)
Comment (Olivernyu) [reply server/src/endpoints/endpoints.py]: Oh yea mb. That import shouldn't be there

**Review by leo4life2 — APPROVED** [review]

**Pull request merged**